File: Backend/ai_security_service/models/security_event.py
import datetime

from sqlalchemy import Column, Integer, String, Float, Enum, DateTime,ForeignKey
from db.db import Base

class SecurityEvent(Base):
    __tablename__ = "security_events"

    id = Column(Integer, primary_key=True, index=True)
    event_type = Column(String, nullable=False)
    attack_type = Column(String, nullable=False)
    email= Column(String, nullable=False)
    summary = Column(String)
    recommendation = Column(String)
    Risk_level = Column(String)
    timestamp = Column(DateTime, default=datetime.datetime.utcnow)
    attempts= Column(Integer, default=1)
    ip_address = Column(String, nullable=True)

    class Config:
        orm_mode = True

File: Frontend/Project_Front/lib/features/Admin/models/security_event.dart
import 'package:flutter/foundation.dart';

class SecurityEvent {
  final int id;
  final String email;
  final String event_type;
  final String attack_type;
  final String summary;
  final String recommendation;
  final String ip_adresse;
  final String risk_level;
  final DateTime timestamp;
  final int? attempts;

  SecurityEvent({
    required this.id,
    required this.email,
    required this.event_type,
    required this.attack_type,
    required this.summary,
    required this.recommendation,
    required this.ip_adresse,
    required this.risk_level,
    required this.timestamp,
    this.attempts,
  });

  factory SecurityEvent.fromJson(Map<String, dynamic> json) {
    return SecurityEvent(
      id: json['id'] ?? 0,
      email: json['email'] ?? '',
      event_type: json['event_type'] ?? '',
      attack_type: json['attack_type'] ?? '',
      summary: json['summary'] ?? '',
      recommendation: json['recommendation'] ?? '',
      ip_adresse: json['ip_adresse'] ?? '',
      risk_level: json['risk_level'] ?? '',
      timestamp: DateTime.parse(json['timestamp']),
      attempts: json['attempts'] ?? 0,
    );
  }
}
